coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -303,21 +317,13 @@ static Schema joinSchemas(@Nonnull Schema leftSchema, @Nonnull Schema rightSchem
     Preconditions.checkNotNull(leftSchema);
     Preconditions.checkNotNull(rightSchema);
 
-    List<Schema.Field> combinedSchemaFields = new ArrayList<>();
-
-    for (Schema.Field field : leftSchema.getFields()) {
-      combinedSchemaFields
-          .add(new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultValue(), field.order()));
-    }
-
-    for (Schema.Field field : rightSchema.getFields()) {
-      combinedSchemaFields
-          .add(new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultValue(), field.order()));
-    }
+    List<Schema.Field> combinedSchemaFields = new ArrayList<>(cloneFieldList(leftSchema.getFields()));
+    combinedSchemaFields.addAll(cloneFieldList(rightSchema.getFields()));
 
     Schema combinedSchema =
         Schema.createRecord(leftSchema.getName(), leftSchema.getDoc(), leftSchema.getNamespace(), leftSchema.isError());
     combinedSchema.setFields(combinedSchemaFields);
+    // Not replicate schema level props as it may not be well-defined in joined schema.
```

Comment:
We may change this to: "While we replicate schema-level props in addPartitionColsToSchema(), we avoid doing this here since it may lead to conflict"

That said, what would be the conflict, and does it warrant throwing an exception instead?

Reply:
@wmoustafa  I actually mean to say the merging of properties from two schema is not a clearly defined behavior, being conflict with each other could be one case. 

Do you want to do something like: merge properties anyway and throw exception if conflict detected?

Reply:
Since you have examples of the properties, you may have a better idea if a conflict means skipping, picking one arbitrarily, or throwing an exception. I think throwing an exception may be a bit extreme.

Reply:
Discussed offline - decide to merge the properties in schema level from left to right.